### PR TITLE
fix(tracing): extend Parity tracing VmExecutedOperation

### LINKED ISSUE #3883: trace_call VmTrace specification
### Describe the bug

I do not know where reth gets the specification for parity style tracing, so this may be a design choice rather than a bug, but I would like some clarification for this issue.

This regards the parity style `trace_call`, specifically `vmTrace`. Reth's `VmExecutedOperation` has the member [push](https://github.com/paradigmxyz/reth/blob/08403c80e210187f1f8e9c9c3d686a789dc73c4a/crates/rpc/rpc-types/src/eth/trace/parity.rs#L311) as an Option
```
pub struct VmExecutedOperation {
    ...
    /// The stack item placed, if any.
    pub push: Option<H256>,
    ...
}
```

whilst both [Parity](https://github.com/openethereum/parity-ethereum/blob/55c90d4016505317034e3e98f699af07f5404b63/ethcore/trace/src/types/trace.rs#L557) and [ethers-rs](https://github.com/gakonst/ethers-rs/blob/030bf439a100dcacd2e968e3114de0612229056b/ethers-core/src/types/trace/mod.rs#L148) implement it as a Vec
```
pub struct VMExecutedOperation {
    ...
    /// The stack item placed, if any.
    pub push: Vec<U256>,
    ...

```
[Erigon](https://github.com/ledgerwatch/erigon/blob/9d3551f8cf3f2835ba8212473024a609a53e24f7/turbo/jsonrpc/trace_adhoc.go#L118) also implements it as a vector:
```
Push  []string      `json:"push"`
```

This breaks compatibility, for example using Reth node as a ethers-rs http provider causes it to fail on deserialization. 

Example of a serialized trace can be seen [here](https://github.com/gakonst/ethers-rs/blob/030bf439a100dcacd2e968e3114de0612229056b/ethers-core/src/types/trace/example-trace-str.rs#L88), vmTrace push should be serialized as:
```
"push": [
    "0x60"
],
```
but is
```
"push": "0x0000000000000000000000000000000000000000000000000000000000000060",
```

Also the [VmInstruction](https://github.com/paradigmxyz/reth/blob/08403c80e210187f1f8e9c9c3d686a789dc73c4a/crates/rpc/rpc-types/src/eth/trace/parity.rs#L293) is missing the `op` and `idx` variables, but ethers-rs has the `op` as non-optional which also breaks deserialization.

### Steps to reproduce

1. Call `trace_call` on a reth node.

### Node logs

_No response_

### Platform(s)

_No response_

### What version/commit are you on?

_No response_

### What database version are you on?

_No response_

### If you've built Reth from source, provide the full command you used

_No response_

### Code of Conduct

- [X] I agree to follow the Code of Conduct

### PR DESCRIPTION
**Pull request by pistomat**
Convert VmExecutedOperation::push from Option<_> to Vec<_> as per ethers-rs and erigon implementation

Closes #3883

### CODE REVIEW

File: crates/rpc/rpc-types/src/eth/trace/parity.rs [hunk 1]
```diff
@@ -327,6 +327,7 @@ pub struct MemoryDelta {
 
 #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
+/// A diff of some storage value.
```
Comment (mattsse) [inline crates/rpc/rpc-types/src/eth/trace/parity.rs:330]: stlye nit: move above derives

**Review by mattsse — COMMENTED** [review]
great, tysm for taking this on.

what's missing?

File: crates/rpc/rpc-types/src/eth/trace/parity.rs (hunk 1 above)
Comment (pistomat) [reply crates/rpc/rpc-types/src/eth/trace/parity.rs:330]: I am also looking into `VMOperation` and members `op` and `idx`.

[Ethers-rs](https://github.com/gakonst/ethers-rs/blob/030bf439a100dcacd2e968e3114de0612229056b/ethers-core/src/types/trace/mod.rs#L137) has it as an enum of OpCodes while `[erigon](https://github.com/ledgerwatch/erigon/blob/9d3551f8cf3f2835ba8212473024a609a53e24f7/turbo/jsonrpc/trace_adhoc.go#L112C2-L113C40)` has both `op` and `idx`.

 I wanted to fix this in one PR, but can make another if you like.

**Comment by pistomat** [comment]
I am also looking into `VMOperation` and members `op` and `idx`.

[Ethers-rs](https://github.com/gakonst/ethers-rs/blob/030bf439a100dcacd2e968e3114de0612229056b/ethers-core/src/types/trace/mod.rs#L137) has only `op` and it as an enum of OpCodes, while [erigon](https://github.com/ledgerwatch/erigon/blob/9d3551f8cf3f2835ba8212473024a609a53e24f7/turbo/jsonrpc/trace_adhoc.go#L112C2-L113C40) has both `op` and `idx`.

 I wanted to fix this in one PR, but can make another if you like.

**Comment by pistomat** [comment]
Ok, adding the `op` and `idx` is work for another time, apperently `vmTrace` is being deprecated on Alchemy and other providers, so seems like there is not much use for this anyway.

**Comment by onbjerg** [comment]
@pistomat If it's work for another time, would you mind creating a new issue for that with a description of what is needed?

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**